core/src/impl/Kokkos_Command_Line_Parsing.cpp

```diff
@@ -241,19 +248,55 @@ void Kokkos::Impl::warn_deprecated_command_line_argument(
 }
 
 namespace {
-std::vector<std::regex> do_not_warn_regular_expressions{
+
+std::vector<std::regex> do_not_warn_command_line_argument_regular_expressions{
     std::regex{"--kokkos-tool.*", std::regex::egrep},
 };
-}
+
+std::vector<std::regex> do_not_warn_environment_variable_regular_expressions{
+    std::regex{"KOKKOS_TOOLS.*", std::regex::egrep},
+    std::regex{"KOKKOS_DIR=.*",
+               std::regex::egrep | std::regex_constants::icase},
+    std::regex{"KOKKOS_HOME=.*",
+               std::regex::egrep | std::regex_constants::icase},
+    std::regex{"KOKKOS_ROOT=.*",
+               std::regex::egrep | std::regex_constants::icase},
+    std::regex{"KOKKOS_PATH=.*",
+               std::regex::egrep | std::regex_constants::icase},
+    std::regex{"KOKKOS_INCLUDE_DIRS=.*",
+               std::regex::egrep | std::regex_constants::icase},
+    std::regex{"KOKKOS_INCLUDE_DIRECTORIES=.*",
+               std::regex::egrep | std::regex_constants::icase},
+    std::regex{"KOKKOS_LIBRARY_DIRS=.*",
+               std::regex::egrep | std::regex_constants::icase},
+    std::regex{"KOKKOS_LIBRARY_DIRECTORIES=.*",
+               std::regex::egrep | std::regex_constants::icase},
+    std::regex{"KOKKOS_SOURCE=.*",
+               std::regex::egrep | std::regex_constants::icase},
+    std::regex{"KOKKOS_SRC=.*",
+               std::regex::egrep | std::regex_constants::icase},
+    std::regex{"KOKKOS_INSTALL=.*",
+               std::regex::egrep | std::regex_constants::icase},
```

Comment:
Let's discuss that list at the next developer meeting.
We need to check with other projects in the Kokkos ecosystem that this works for them.

I purposefully omitted `KOKKOS_VARIORUM_FUNC_TYPE`, I don't think we should accommodate this one.